test/core/integration/concrete/lockup-base/withdraw-multiple/withdrawMultiple.t.sol

```diff
@@ -184,9 +184,10 @@ contract WithdrawMultiple_Integration_Concrete_Test is Integration_Test {
         givenNoNullStreams
         givenNoDEPLETEDStreams
         whenNoZeroAmounts
+        whenCallerAuthorizedForAllStreams
```

Comment:
it does not matter, in your singletone test PR, you removed it while this has dedicated logic to run the tests with the `msg.sender` as these:

https://github.com/sablier-labs/v2-core/blob/5764dd34aa56f544ff1e5caaa0be00bf9f9a610e/test/core/integration/concrete/lockup-base/withdraw-multiple/withdrawMultiple.t.sol#L162-L165

otherwise we can remove it as it is not used

BTT is incomplete, the main purpose of it, is to simply divide in binary branches the logic.

Reply:
> I would recommend you to start a discussion on ideas to improve it.

that's a good idea, but it takes time, and we need to take a decision to not stop our dev proces

---

also, why does it make not Bulloak compatible if we include other modifiers (that have unique logic) to the function tests? 
```solidity
// BTT modifier
modifier whenX {
   _; 
}
// non BTT modifier 
modifier whenAllCallersPossible {
    // run test with caller 1
    _;
    // run test in caller 2
    _;
    // run test in caller 3
    _;
}

function testFoo_SomeY() public whenX  whenAllCallersPossible {
    assertFalse(Some(condition));
    assertTrue(Some(otherCondition));
}
```

Reply:
Alongside `whenAllCallersPossible ` there would also be a case when all callers are not possible. A realistic scenario would be

```tree
Doge_Test
├── when not x
│   └── it should revert
└── when x
    ├── when one caller is not authorized
    │   └── it should revert
    └── when all callers are authorized
        └── when some Y
            └── it do somethin
```

for which bulloak would generate 

```solidity
contract Doge_Test {
    function test_RevertWhen_NotX() external {
        // it should revert
    }

    modifier whenX() {
        _;
    }

    function test_RevertWhen_OneCallerIsNotAuthorized() external whenX {
        // it should revert
    }

    modifier whenAllCallersAreAuthorized() {
        _;
    }

    function test_WhenSomeY() external whenX whenAllCallersAreAuthorized {
        // it do somethin
    }
}
```

in which you can override `whenAllCallersPossible`

Does that make sense?

Reply:
> in which you can override whenAllCallersPossible
Does that make sense?

Yes, it makes sense, but I believe you are missing the fact that the caller tree branches are covered in the `withdraw.tree` tests (which we don’t need to specifically re-test here). Here, we are talking about `withdrawMultiple`. By addind the `whenAllCallersPossible` we just adding "nice have" elegant (without being redundant), that is not a requirement

---

This question was not a question; could you answer it, please? Why does it make it not compatible with Bulloak if we include other modifiers (that have unique logic) in the function tests?

Reply:
> also, why does it make not Bulloak compatible if we include other modifiers (that have unique logic) to the function tests?

Even though bulloak check would pass since it ignored additional modifiers, the goal with the bulloak and BTT is to write tree first and then generate the test contract automatically. The automation also adds required modifiers. If we add a modifier to the test that is not generated automatically by the bulloak, I would consider the test to be incompatible with the tree file.

Imagine a modifier is added but not a part of the tree and that modifier has some logic inside, this can introduce errors to the tests.

Reply:
> the goal with the bulloak and BTT is to write tree first and then generate the test contract automatically.

yes, but we are _not_ achieving this since we have agreed to skip the modifiers since we had a lot of duplicated modifiers in each file

> The automation also adds required modifiers

yes, which we ignore since we have the `Modifiers` file

> I would consider the test to be incompatible with the tree file

by what you said above, even the current version is incompatible


> Imagine a modifier is added but not a part of the tree and that modifier has some logic inside, this can introduce errors to the tests.

for sure if it adds an error to the test, we would catch this when we run `forge test` 
as said above: "By adding the whenAllCallersPossible we just add a "nice have" in an elegant way (without being redundant), that is not a requirement"